kombu/asynchronous/aws/sqs/connection.py

```diff
@@ -25,6 +29,75 @@ def __init__(self, sqs_connection, debug=0, region=None, **kwargs):
             **kwargs
         )
 
+    def _create_query_request(self, operation, params, queue_url, verb, callback=None):  # noqa
+        params = params.copy()
+        if operation:
+            params['Action'] = operation
+
+        # defaults for non-get
+        param_payload = {'data': params}
+        if verb.lower() == 'get':
+            # query-based opts
+            param_payload = {'params': params}
+
+        return AWSRequest(method=verb, url=queue_url, **param_payload)
+
+    def _create_json_request(self, operation, params, queue_url):
+        params = params.copy()
+        params['QueueUrl'] = queue_url
+
+        service_model = self.sqs_connection.meta.service_model
+        operation_model = service_model.operation_model(operation)
+
+        url = self.sqs_connection._endpoint.host
+
+        headers = {}
+        # Content-Type
+        json_version = operation_model.metadata['jsonVersion']
+        content_type = f'application/x-amz-json-{json_version}'
+        headers['Content-Type'] = content_type
+
+        # X-Amz-Target
+        target = '{}.{}'.format(
+            operation_model.metadata['targetPrefix'],
+            operation_model.name,
+        )
+        headers['X-Amz-Target'] = target
+
+        param_payload = {
+            'data': json.dumps(params),
+            'headers': headers
+        }
+
+        method = operation_model.http.get('method', Serializer.DEFAULT_METHOD)
+        return AWSRequest(
+            method=method,
+            url=url,
+            **param_payload
+        )
+
+    def make_request(self, operation, params, queue_url, verb, callback=None):  # noqa
+        signer = self.sqs_connection._request_signer
+
+        service_model = self.sqs_connection.meta.service_model
+        protocol = service_model.protocol
+
+        if protocol == 'query':
+            request = self._create_query_request(
+                operation, params, queue_url, verb)
+        elif protocol == 'json':
+            request = self._create_json_request(operation, params, queue_url)
+        else:
+            raise Exception(f'Unsupported protocol: {protocol}.')
+
+        signing_type = 'presignurl' if request.method.lower() == 'get' \
```

Comment:
Why use `presignurl` on `get`?  It's not clear from https://github.com/boto/botocore/blob/develop/botocore/signers.py when pre-signed urls should be used.  Can you provide some documentation?

Reply:
It also confused me, and the only explanation I have is that the documentation [here](https://github.com/boto/botocore/blob/develop/botocore/signers.py#L132) is inaccurate and they meant to instead say:

> This should be used when pre-signing a GET* request"

Having said that, I believe this code (which I copied from the [original implementation of `make_request`](https://github.com/celery/kombu/blob/main/kombu/asynchronous/aws/connection.py#L194-L196) in the `kombu.aws.connection.AsyncAWSQueryConnection` class) is actually incorrect, since the signer name is `presign-url` rather than `presignurl`, but I was planning on keeping it as is until I get feedback from the botocore team.